integration_tests/19_limit_runs_per_branch.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+my_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "$my_dir/header.sh"
+
+testground plan import --from ./plans --name testground
+
+pushd $TEMPDIR
+
+testground build single \
+    --plan testground/placebo \
+    --builder docker:go \
+    --wait | tee build.out
+export ARTIFACT=$(awk -F\" '/generated build artifact/ {print $8}' build.out)
+docker tag $ARTIFACT testplan:placebo
+
+testground healthcheck --runner local:docker --fix
+
+# Run first test case, but do not --wait
+# We must start the 2nd case at the same time
+t1=$(testground run single \
+    --plan=testground/placebo \
+    --testcase=ok \
+    --builder=docker:go \
+    --use-build=testplan:placebo \
+    --runner=local:docker \
+    --instances=1 \
+    --metadata-branch b1 \
+    --metadata-repo r1 \
+    --collect)
+
+# Very unwieldy, $59 is the ID in the log
+run_id1=$(echo $t1 | awk '/run is queued with ID: / {print $59}')
+
+# Run the second test case, and wait for it to finish
+t2=$(testground run single \
+    --plan=testground/placebo \
+    --testcase=ok \
+    --builder=docker:go \
+    --use-build=testplan:placebo \
+    --runner=local:docker \
+    --instances=1 \
+    --metadata-branch b1 \
+    --metadata-repo r1 \
+    --wait \
+    --collect)
+
+run_id2=$(echo $t2 | awk '/run is queued with ID: / {print $59}')
+
+# First run must be canceled
+check_testground_task_status "$run_id1" 'canceled'
+# Second must succeed
+check_testground_task_status "$run_id2" 'success'
```

Comment:
👍 

I thought we couldn't cancel a task that is already running,
does it mean that Testground is so slow to pick up tasks that we haven't started `task1` when we schedule `task2`?

Reply:
That's correct. The time interval between requesting the two runs  is small enough that the supervisor does not yet start `task1`.

The supervisor sleeps whenever the queue is empty:
https://github.com/testground/testground/blob/775a58896b01063ae7d9fef0afbb4faf7a5f2bf5/pkg/engine/supervisor.go#L58

```
tsk, err := e.queue.Pop()
if err == task.ErrQueueEmpty {
	time.Sleep(time.Second)
	continue
}
```
(I'm not pleased with the way this test is setup at the moment, and we should consider making the sleep window configurable)

Reply:
👍 That might be flaky, let's merge as is but double-check the CI when it fails.